Minio.Functional.Tests/FunctionalTest.cs

```diff
@@ -1393,36 +1389,38 @@ internal async static Task CopyObject_Test2(MinioClient minio)
+                throw ex;
+            }
+
+            try
+            {
                 CopyConditions conditions = new CopyConditions();
                 conditions.SetMatchETag("TestETag");
-                try
-                {
-                    CopySourceObjectArgs copySourceObjectArgs = new CopySourceObjectArgs()
-                                                                            .WithBucket(bucketName)
-                                                                            .WithObject(objectName)
-                                                                            .WithCopyConditions(conditions);
-                    CopyObjectArgs copyObjectArgs = new CopyObjectArgs()
-                                                                .WithCopyObjectSource(copySourceObjectArgs)
-                                                                .WithBucket(destBucketName)
-                                                                .WithObject(destObjectName);
-
-                    await minio.CopyObjectAsync(copyObjectArgs);
-                }
-                catch (MinioException ex)
-                {
-                    Assert.IsTrue(ex.Message.Contains("MinIO API responded with message=At least one of the pre-conditions you specified did not hold"));
-                }
-                catch (Exception ex)
-                {
-                    new MintLogger(nameof(CopyObject_Test2), copyObjectSignature, "Tests whether CopyObject with Etag mismatch passes", TestStatus.FAIL, (DateTime.Now - startTime), ex.Message, ex.ToString(), args:args).Log();
-                    throw ex;
-                }
+                CopySourceObjectArgs copySourceObjectArgs = new CopySourceObjectArgs()
+                                                                        .WithBucket(bucketName)
+                                                                        .WithObject(objectName)
+                                                                        .WithCopyConditions(conditions);
+                CopyObjectArgs copyObjectArgs = new CopyObjectArgs()
+                                                            .WithCopyObjectSource(copySourceObjectArgs)
+                                                            .WithBucket(destBucketName)
+                                                            .WithObject(destObjectName);
 
+                await minio.CopyObjectAsync(copyObjectArgs);
+            }
+            catch (MinioException ex)
+            {
+                Assert.IsTrue(ex.Message.Contains("MinIO API responded with message=At least one of the pre-conditions you specified did not hold"));
                 new MintLogger("CopyObject_Test2", copyObjectSignature, "Tests whether CopyObject with Etag mismatch passes", TestStatus.PASS, (DateTime.Now - startTime), args:args).Log();
             }
```

Comment:
missing throw in catch

Reply:
is the Assert code needed here?

Reply:
> is the Assert code needed here?

Yes. Checks for pre-condition, specifically.

Reply:
I am not sure if this is correct. 
```
Assert.IsTrue(ex.Message.Contains("MinIO API responded with message=At least one of the pre-conditions you specified did not hold"))
```

Reply:
OK, changed.